### PR TITLE
Improve code coverage for algorithm/predicates.

### PR DESCRIPTION
**Pull request by k82cn**
Improve code coverage for algorithm/predicates for #39559 .

Improved coverage from 71.3% to 81.9%.

Coverage report: [combined-coverage.html.gz](https://github.com/kubernetes/kubernetes/files/691518/combined-coverage.html.gz)

### CODE REVIEW

**Comment by k82cn** [comment]
@k8s-bot bazel test this

**Comment by davidopp** [comment]
Unfortunately I won't have time to review this; hopefully someone else from @kubernetes/sig-scheduling-misc can do it.

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/kubernetes/pkg/api/resource"
 	"k8s.io/kubernetes/pkg/api/v1"
+	metav1 "k8s.io/kubernetes/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: I'm not a huge fan of introducing a dep on the api-cache to improve coverage.  Currently there are ongoing efforts to eliminate unnecessary dependencies in the code such that algorithm could be lifted to be used externally.  I preferred the FakeNodeList tbh.

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 2]
```diff
@@ -2817,12 +2824,72 @@ func TestInterPodAffinityWithMultipleNodes(t *testing.T) {
 			},
 			test: "NodeA and nodeB have same topologyKey and label value. NodeA has an existing pod that match the inter pod affinity rule. The pod can not be scheduled onto nodeA and nodeB but can be schedulerd onto nodeC",
 		},
+		{
+			pod: &v1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"foo": "123"},
+					Annotations: map[string]string{
+						v1.AffinityAnnotationKey: `
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: This is no longer how affinity works, it's a field now, and annotation has been removed.

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 3]
```diff
@@ -3335,3 +3409,201 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+			Node: &v1.Node{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "host1",
+				},
+			},
+			Fits: true,
+		},
+		{
+			Name: "label zone failure domain mached",
+			Pod: &v1.Pod{
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: b/c most of this code repeats, consider a utility f(n).  We did a similar change in the affinity changes recently.

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go (hunk 2 above)
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: wow, yes; and I think we should also remove `v1.AffinityAnnotationKey` from the code.

**Comment by k82cn** [comment]
@timothysc , moved podAffity related test to #39993 ; only keep `VolumeZone` related test.

**Comment by k82cn** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by k82cn** [comment]
@timothysc , updated to address review comments accordingly :).

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 4]
```diff
@@ -3335,3 +3335,149 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+			Pod:  createPodWithVolume("pod_1", "vol_1", "PVC_1"),
+			Node: &v1.Node{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "host1",
+				},
+			},
+			Fits: true,
+		},
+		{
+			Name: "label zone failure domain mached",
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: s/mached/matched

**Comment by timothysc** [comment]
There are some minor spelling things, but I'll fix them in another PR.  
Thx @k82cn! 
/lgtm

**Comment by k82cn** [comment]
@k8s-bot bazel test this

**Comment by k82cn** [comment]
The build failed because of PR at bcde05753b04e94ee5d9c1d2583e5a90dbe3ced3 , I'm going to update PR accordingly.

**Comment by k82cn** [comment]
@timothysc , the code was updated to align with  bcde057 ; PTAL : updated `v1.ObjectMeta` to `metav1.ObjectMeta`.

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 5]
```diff
@@ -3437,3 +3437,149 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+			Node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "host1",
+					Labels: map[string]string{metav1.LabelZoneFailureDomain: "zone_1", "uselessLabel": "none"},
+				},
+			},
+			Fits: true,
+		},
+		{
+			Name: "label zone region mached",
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: matched

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 6]
```diff
@@ -3437,3 +3437,149 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+			Node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "host1",
+					Labels: map[string]string{metav1.LabelZoneRegion: "zone_2", "uselessLabel": "none"},
+				},
+			},
+			Fits: true,
+		},
+		{
+			Name: "label zone region dismached",
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: failed match.

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 7]
```diff
@@ -3437,3 +3437,149 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+			Node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "host1",
+					Labels: map[string]string{metav1.LabelZoneRegion: "no_zone_2", "uselessLabel": "none"},
+				},
+			},
+			Fits: false,
+		},
+		{
+			Name: "label zone failure domain dismached",
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: failed match

**Comment by timothysc** [comment]
@k82cn it didn't go through last night, could you please update some of the misspellings and I'll lgtm right after.

**Pull request merged**